tests/unit/backend/layers/business/test_business.py

```diff
@@ -168,22 +175,33 @@ def test_create_collection_with_valid_doi_ok(self):
         self.assertIsNotNone(collection_from_database.publisher_metadata)
         self.assertEqual(collection_from_database.publisher_metadata, expected_publiser_metadata)
 
-    def test_create_collection_with_invalid_doi_fail(self):
+    def test_create_collection_with_not_found_doi_fail(self):
         """
-        A collection with an invalid DOI will not be created
+        A collection with a DOI that cannot be found on Crossref will not be created
         """
         links_with_doi = [
             Link("test doi", "doi", "http://bad.doi")
         ]
 
         self.sample_collection_metadata.links = links_with_doi
 
-        # TODO: make sure that we don't need different actions depending on which exception
-        self.crossref_provider.fetch_metadata = Mock(side_effect=CrossrefException("Error!"))
+        self.crossref_provider.fetch_metadata = Mock(side_effect=CrossrefDOINotFoundException("Error!"))
```

Comment:
We definitely needed more behaviors. I split this tests in two, each one corresponding to a different type of Crossref exception.